app/frontend/actions/wallet.ts

```diff
@@ -182,18 +183,25 @@ export default (store: Store) => {
     return true
   }
 
-  const reloadWalletInfo = async (state: State) => {
+  let lastReloadTime = new Date()
+  const reloadWalletInfo = async (state: State): Promise<void> => {
```

Comment:
I would give this function a parameter for overriding the time condition. I think we should reload wallet info after transaction submission no matter what. In case a user reloads wallet and then sends tx under 30 seconds his balances and tx history would not update.